core/src/main/java/io/questdb/std/Files.java

```diff
@@ -62,46 +62,29 @@ public final class Files {
     public static final int WINDOWS_ERROR_FILE_EXISTS = 0x50;
     private static final AtomicInteger OPEN_FILE_COUNT = new AtomicInteger();
     private static final int VIRTIO_FS_MAGIC = 0x6a656a63;
+    private static final AtomicInteger fdCounter = new AtomicInteger();
+    private static final LongHashSet openFds = new LongHashSet();
+    // To be set in tests to check every call for using OPEN file descriptor
+    public static boolean PARANOIA_FD_MODE = false;
     public static boolean VIRTIO_FS_DETECTED = false;
-    static IntHashSet openFds;
 
     private Files() {
         // Prevent construction.
     }
 
-    public native static boolean allocate(int fd, long size);
-
-    public native static long append(int fd, long address, long len);
-
-    public static synchronized boolean auditClose(int fd) {
-        if (fd < 0) {
-            throw new IllegalStateException("Invalid fd " + fd);
-        }
-        if (openFds.remove(fd) == -1) {
-            throw new IllegalStateException("fd " + fd + " is already closed!");
-        }
-        return true;
+    public static boolean allocate(long fd, long size) {
+        return allocate(toOsFd(fd), size);
     }
 
-    public static synchronized boolean auditOpen(int fd) {
-        if (openFds == null) {
-            openFds = new IntHashSet();
-        }
-        if (fd < 0) {
-            throw new IllegalStateException("Invalid fd " + fd);
-        }
-        if (openFds.contains(fd)) {
-            throw new IllegalStateException("fd " + fd + " is already open");
-        }
-        openFds.add(fd);
-        return true;
+    public static long append(long fd, long address, long len) {
+        return append(toOsFd(fd), address, len);
     }
 
-    public static int bumpFileCount(int fd) {
+    public static long bumpFileCount(int fd) {
```

Comment:
This method now creates and returns the unique FD, which seems to be its main purpose. It also bumps the open file count as a side effect.

I'd suggest a name that relates more to that primary purpose: `createUniqueFd()` or similar.

Reply:
done in 5f22804767ebade94ec0efa41bd2ce172cc78160